sdk/openai/Azure.AI.OpenAI/src/Custom/FineTuning/AzureFineTuningClient.Extensions.cs

```diff
@@ -0,0 +1,100 @@
+/// </summary>
+internal static class AzureFineTuningClientExtensions
+{
+    internal static PipelineMessageClassifier PipelineMessageClassifier { get; } =
+            PipelineMessageClassifier.Create(stackalloc ushort[] { 204 });
+
+    /// <summary>
+    /// Deletes an Azure OpenAI fine tuning job.
+    /// </summary>
+    /// <param name="client">The Azure OpenAI fine tuning client.</param>
+    /// <param name="jobId">The identifier for the fine tuning job to delete.</param>
+    /// <param name="options">(Optional) The request options to use.</param>
+    /// <returns>True if the job deletion request was sent successfully, false otherwise.</returns>
+    /// <remarks>The Azure OpenAI service will always return a success (HTTP 204) regardless of whether or not
+    /// the job you are trying to delete exists.</remarks>
+    [Experimental("AOAI001")]
+    public static bool DeleteJob(this FineTuningClient client, string jobId, RequestOptions? options = null)
+#pragma warning disable AZC0106 // Non-public asynchronous method needs 'async' parameter.
+        => DeleteJobSyncOrAsync(client, false, jobId, options).EnsureCompleted();
+#pragma warning restore AZC0106 // Non-public asynchronous method needs 'async' parameter.
+
+    /// <summary>
+    /// Deletes an Azure OpenAI fine tuning job.
+    /// </summary>
+    /// <param name="client">The Azure OpenAI fine tuning client.</param>
+    /// <param name="jobId">The identifier for the fine tuning job to delete.</param>
+    /// <param name="options">(Optional) The request options to use.</param>
+    /// <returns>True if the job deletion request was sent successfully, false otherwise.</returns>
+    /// <remarks>The Azure OpenAI service will always return a success (HTTP 204) regardless of whether or not
+    /// the job you are trying to delete exists.</remarks>
+    [Experimental("AOAI001")]
+    public static Task<bool> DeleteJobAsync(this FineTuningClient client, string jobId, RequestOptions? options = null)
+        => DeleteJobSyncOrAsync(client, true, jobId, options).AsTask();
+
+    /// <summary>
+    /// Deletes an Azure OpenAI fine tuning job.
+    /// </summary>
+    /// <param name="client">The Azure OpenAI fine tuning client.</param>
+    /// <param name="isAsync">True if this should be an async network request, false otherwise/</param>
+    /// <param name="jobId">The identifier for the fine tuning job to delete.</param>
+    /// <param name="options">(Optional) The request options to use.</param>
+    /// <returns>True if the job deletion request was sent successfully, false otherwise.</returns>
+    /// <remarks>The Azure OpenAI service will always return a success (HTTP 204) regardless of whether or not
+    /// the job you are trying to delete exists.</remarks>
+    [Experimental("AOAI001")]
+    internal static async ValueTask<bool> DeleteJobSyncOrAsync(this FineTuningClient client, bool isAsync, string jobId, RequestOptions? options = null)
```

Comment:
I had no idea that deleting jobs was unique to the Azure OpenAI specification! Do you know if it works on OpenAI and is just undocumented?

Separately from that: why an extension method vs. just a plain additive one?

Reply:
Yes seems to be unique to Azure OpenAI. In the OpenAI world you'd use the model client for this. But in Azure OpenAI you do it via the fine tuning APIs. It is an extension method since I didn't want to add this OpenAI fine tuning client, and the AzureFineTuningClient is internal

Reply:
Ah, right, of course -- I keep forgetting that we don't have the scenario clients public.

Reply:
We can discuss this making a public extension method for only Azure OpenAI in a future release. For now since this class is internal, it is only accessible to us and the test code.